internals/overlord/checkstate/manager.go

```diff
@@ -359,40 +348,37 @@ func (m *CheckManager) updateCheckInfo(config *plan.Check, changeID string, fail
 		startup = plan.CheckStartupEnabled
 	}
 
-	if check, ok := m.checks[config.Name]; ok {
-		check.Level = config.Level
-		check.Startup = startup
-		check.Status = status
-		check.Failures = failures
-		check.Threshold = config.Threshold
-		check.ChangeID = changeID
-	} else {
-		m.checks[config.Name] = &CheckInfo{
-			Name:              config.Name,
-			Level:             config.Level,
-			Startup:           startup,
-			Status:            status,
-			Failures:          failures,
-			Threshold:         config.Threshold,
-			ChangeID:          changeID,
-			PerformCheckCount: 0,
-			RecoverCheckCount: 0,
-		}
+	check, ok := m.checks[config.Name]
+	if !ok {
+		check = &CheckInfo{Name: config.Name}
+		m.checks[config.Name] = check
 	}
+	check.Level = config.Level
+	check.Startup = startup
+	check.Status = status
+	check.Failures = failures
+	check.Threshold = config.Threshold
+	check.ChangeID = changeID
 }
 
 func (m *CheckManager) incPerformCheckCount(config *plan.Check) {
 	m.checksLock.Lock()
 	defer m.checksLock.Unlock()
 
-	m.checks[config.Name].PerformCheckCount += 1
+	check, ok := m.checks[config.Name]
```

Comment:
We should do this check in `incRecoverCheckCount` as well. Maybe we can have a little helper method `ensureCheck(name string) *CheckInfo` that does this dance, to avoid repeating it three times.